applications/dashboard/src/scripts/components/DashboardHeaderBlock.tsx

```diff
@@ -35,4 +30,4 @@ function DashboardHeaderBlock(props: IProps) {
     );
 }
 
-export default withRouter(DashboardHeaderBlock);
+export default withRouter(DashboardHeaderBlock as any); // Typescript is complaining that RouteComponentProps are optional, but they are if you don't import the default.
```

Comment:
We can resolve this pretty easily actually, now that we have the latest version of react router.

They have a hook.

```ts
    const history = useHistory();
	// Use this instead of through props.
```

Then we can remove the `withRouter()` wrapper and the `extends RouteComponentProps`.